src/autonomous/AutonomousTask.cpp

```diff
@@ -0,0 +1,57 @@
+#include "AutonomousTask.h"
+
+#include "../Constants.h"
+#include "../Globals.h"
+#include "../commands/DriveToWaypointCommand.h"
+#include "../world_interface/world_interface.h"
+
+namespace autonomous {
+
+AutonomousTask::AutonomousTask(){};
+
+AutonomousTask::~AutonomousTask() {
+	if (_autonomous_task_thread.joinable()) {
+		_autonomous_task_thread.join();
+	}
+}
+
+void AutonomousTask::start(const navtypes::pose_t& waypointCoords) {
+	setWaypoint(waypointCoords);
+	_kill_called = false;
+	_autonomous_task_thread = std::thread(&autonomous::AutonomousTask::navigate, this);
+}
+
+void AutonomousTask::navigate() {
+	commands::DriveToWaypointCommand cmd(_waypoint_coords, THETA_KP, DRIVE_VEL, SLOW_DRIVE_VEL,
+										 DONE_THRESHOLD, CLOSE_TO_TARGET_DUR_VAL);
+
+	while (!cmd.isDone()) {
+		auto latestGPS = robot::readGPS();
+		if (latestGPS.isFresh(2000ms)) {
+			cmd.setState(latestGPS, robot::types::dataclock::now());
```

Comment:
`latestGPS` is a `point_t`, so it has `[x, y, 1]`, but we want a `pose_t`, which has `[x, y, theta]`. Confusingly, they're both `Eigen::Vector3d`, so the compiler doesn't complain.

GPS only gives position information, so you'll need to get heading from the IMU. Call `robot::readIMUHeading()` to get that and assemble that into a single `pose_t`.

Reply:
Would appreciate if you could double check this carefully. I think I got it but not sure.